src/modules/mavlink/mavlink_main.cpp

```diff
@@ -1448,17 +1448,6 @@ Mavlink::task_main(int argc, char *argv[])
 	_datarate = 0;
 	_mode = MAVLINK_MODE_NORMAL;
 
-#ifdef __PX4_NUTTX
-	/* the NuttX optarg handler does not
-	 * ignore argv[0] like the POSIX handler
-	 * does, nor does it deal with non-flag
-	 * verbs well. Remove the application
-	 * name and the verb.
-	 */
-	argc -= 3;
-	argv += 3;
-#endif
-
```

Comment:
@kd0aij 

Is 3 correct? After @Inspirati 's nsh parser fix I would have expected it to be the same as what is on master https://github.com/PX4/Firmware/blob/master/src/modules/mavlink/mavlink_main.cpp#L1501-L1509 . Did you try 2?